Assets/MRTK/Core/Providers/InputSimulation/SimulatedMotionControllerDataProvider.cs

```diff
@@ -0,0 +1,145 @@
+        {
+            IsSelecting = false;
+            IsGrabbing = false;
+            IsPressingMenu = false;
+        }
+
+        internal MixedRealityPose UpdateControllerPose()
+        {
+            Vector3 screenPosition = CameraCache.Main.ViewportToScreenPoint(ViewportPosition);
+            Vector3 worldPosition = CameraCache.Main.ScreenToWorldPoint(screenPosition + JitterOffset);
+
+            Quaternion localRotation = Quaternion.Euler(ViewportRotation);
+            Quaternion worldRotation = CameraCache.Main.transform.rotation * localRotation;
+
+            return new MixedRealityPose(worldPosition, worldRotation);
+        }
+
+    }
+
+    /// <inheritdoc />
+    public class SimulatedMotionControllerDataProvider : SimulatedControllerDataProvider
+    {
+        // Cached delegates for position and rotation update
+        private SimulatedMotionControllerData.MotionControllerPoseUpdater updaterLeft;
+        private SimulatedMotionControllerData.MotionControllerPoseUpdater updaterRight;
+
+        public SimulatedMotionControllerDataProvider(MixedRealityInputSimulationProfile _profile) : base(_profile)
+        {
+            InputStateLeft = new SimulatedMotionControllerState(Handedness.Left);
+            InputStateRight = new SimulatedMotionControllerState(Handedness.Right);
+        }
+
+        /// <inheritdoc />
+        internal override void SimulateInput(ref long lastMotionControllerTrackedTimestamp, SimulatedControllerState state, bool isSimulating, bool isAlwaysVisible, MouseDelta mouseDelta, bool useMouseRotation)
+        {
+            var motionControllerState = state as SimulatedMotionControllerState;
+            if (motionControllerState == null)
+            {
+                return;
+            }
+            bool enableTracking = isAlwaysVisible || isSimulating;
+            if (!motionControllerState.IsTracked && enableTracking)
+            {
+                ResetInput(motionControllerState, isSimulating);
+            }
+            if (isSimulating)
+            {
+                motionControllerState.SimulateInput(mouseDelta, useMouseRotation, profile.MouseRotationSensitivity, profile.MouseControllerRotationSpeed, profile.ControllerJitterAmount);
+
+                motionControllerState.IsSelecting = KeyInputSystem.GetKey(profile.MotionControllerTriggerKey);
+                motionControllerState.IsGrabbing = KeyInputSystem.GetKey(profile.MotionControllerGrabKey);
+                motionControllerState.IsPressingMenu = KeyInputSystem.GetKey(profile.MotionControllerMenuKey);
+            }
+
+            // Update tracked state of a motion controller.
+            // If hideTimeout value is null, motion controllers will stay visible after tracking stops.
+            // TODO: DateTime.UtcNow can be quite imprecise, better use Stopwatch.GetTimestamp
+            // https://stackoverflow.com/questions/2143140/c-sharp-datetime-now-precision
+            DateTime currentTime = DateTime.UtcNow;
```

Comment:
Is the plan to switch to Stopwatch.GetTimestamp or keep DateTime.UtcNow?

Reply:
The code here is from the existing `SimulatedHandDataProvider` and there is no plan to change in this PR.